src/bun.js/net.exports.js

```diff
@@ -53,9 +53,255 @@ export function isIP(s) {
   return 0;
 }
 
+const { Bun, createFIFO } = import.meta.primordials;
+const { connect: bunConnect } = Bun;
+const { Duplex } = import.meta.require("node:stream");
+
+export class Socket extends Duplex {
+  static #Handlers = {
+    close: Socket.#Close,
+    data(socket, buffer) {
+      const self = socket.data;
+      self.bytesRead += buffer.length;
+      if (self.#readQueue.isEmpty()) {
+        if (self.push(buffer)) return;
```

Comment:
Do we return Buffer objects in the stream class? We could add a flag to do that here because IIRC, node:stream will internally create a Buffer object

Reply:
Technically `Uint8Array` (so basically `Buffer` from Node.js PoV) &minus; will investigate if `node:stream` does extra allocation or not :thinking:

Reply:
yeah it does in the streams code when you push a Uint8Array, it creates a new Buffer